awx/main/models/notifications.py

```diff
@@ -233,6 +236,8 @@ def send_notification_templates(self, status_str):
         if notification_templates:
             if status_str == 'succeeded':
                 notification_template_type = 'success'
+            elif status_str == 'running':
+                notification_template_type = 'started'
             else:
                 notification_template_type = 'error'
             all_notification_templates = set(notification_templates.get(notification_template_type, []) + notification_templates.get('any', []))
```

Comment:
I could see an argument for *not* including the `_any` relationship when `status_str="running"`.  If you think about what "any" means today, it really means "at the end, regardless of success or failure".

If you kept this code as-is, and had an "any" template defined, it would run *twice*.

1.  When the job went into `running` state, it would run.
2. When the job finished and went into either `success` or `failed` it would run.

Maybe we should change this logic to something like:

```python
if status_str == 'running':
    all_notification_templates = notification_templates.get('started', [])
else:
    if status_str == 'succeeded':
        notification_template_type = 'success'
    else:
        notification_template_type = 'error'
    all_notification_templates = set(notification_templates.get(notification_template_type, []) + notification_templates.get('any', []))
```

Reply:
Note that 'any' is currently an API-only construct that's not in the UI. TBD how much it's used in practice.

Is it worth migrating 'any' to 'finished'?

Reply:
Eh?  In another PR if we wanted to.  For now I'd prefer to just avoid _behavioral_ changes.